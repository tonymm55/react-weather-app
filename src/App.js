// import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="Prop">
      <header className="App-header">
        <h1>Weather App Header from App.js</h1>
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          My Weather Application      
        </a>
      </header>
    </div>
  );
}

export default App;
