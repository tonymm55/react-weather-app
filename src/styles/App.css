.weather-app {
  margin: 0 20px;
}

.forecast-details {
  margin-top: 100px;
  margin-bottom: 20px;
  text-align: center;
}

.forecast-details__maxTemperature {
  font-size: 15px;
  margin-top: 10px;
  font-weight: bold;
}

.forecast-details__minTemperature {
  font-size: 15px;
  margin-top: 10px;
  font-weight: bold;
}

.forecast-details__wind {
  font-size: 15px;
  margin-top: 10px;
  font-weight: bold;
}

.forecast-details__humidity {
  font-size: 15px;
  margin-top: 10px;
  font-weight: bold;
}

.forecast-details__date {
  font-weight: bold;
  font-size: 1.5rem; 
}


.App-header {
  background-color: #282c34;
  /* background-color: #eceef1; */
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  font-family: Orbitron;
}

.App-link {
  color: #61dafb;
}

.search-form {
  padding-bottom: 20px;
}

.search-form button {
  display: inline-block;
  padding: .1rem .1rem;
  background-color: black;
  color: white;
  font-size: 1.rem;
  text-align: center;
  text-decoration: none;
  transition: all 0.3s ease;
}

.search-form button:hover {
  background-color: white;
  color: black;
}

.search-form button:active {
  background-color: white;
  color: red;
}
