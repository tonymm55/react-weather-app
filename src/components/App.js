/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable react/jsx-filename-extension */
import "../styles/App.css";
import LocationDetails from "./LocationDetails";


function App(props) {
  const { location } = props;
  return (
    <div className="App">
      <LocationDetails city={location.city} country={location.country} />
    </div>
  );
}

        <header className="App-header">
          <h1>Weather App</h1>
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            My Weather Application
          </a>
        </header>
      </div>
  );
}

export default App;

