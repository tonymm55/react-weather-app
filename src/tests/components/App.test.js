import { render, screen, waitFor } from "@testing-library/react";
import App from "../../components/App";
import forecast from "../../data/forecast.json";

describe("App", () => {
  it("renders without errors", () => {
    expect(() => render(<App location={forecast.location} forecasts={forecast.forecasts} />)).not.toThrow();
  });
});

describe("App", () => {
  it("renders correctly", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("App", () => {
  it("renders the app components", () => {
    render(<App location={forecast.location} forecasts={forecast.forecasts} />);
    const locationDetails = screen.getByTestId("location-details");
    const searchForm = screen.getByTestId("search-form");
    const forecastSummaries = screen.getByTestId("forecast-summaries");
    const forecastDetails = screen.getByTestId("forecast-details");
    expect(locationDetails).toBeInTheDocument();
    expect(searchForm).toBeInTheDocument();
    expect(forecastSummaries).toBeInTheDocument();
    expect(forecastDetails).toBeInTheDocument();
  });

  it("renders the correct number of ForecastSummaries components", () => {
    render(<App location={forecast.location} forecasts={forecast.forecasts} />);
    const forecastSummaries = screen.getAllByTestId("forecast-summaries");
    expect(forecastSummaries).toHaveLength(1);
  });
});

describe("SearchForm", () => {
  it("matches snapshot", () => {
    const { searchForm } = render(<App />);
    expect(searchForm).toMatchSnapshot();
  });

  it("renders the correct text for the search button", () => {
    render(<App location={forecast.location} forecasts={forecast.forecasts} />);
    const searchButton = screen.getByRole("button");
    expect(searchButton).toHaveTextContent("Search Location");
  });
});
