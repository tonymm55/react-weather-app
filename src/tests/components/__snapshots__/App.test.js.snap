// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="weather-app"
>
  <div
    class="location-details"
    data-testid="location-details"
  >
    <h1
      class="location-details"
    >
      , 
    </h1>
  </div>
  <div
    class="search-form"
    data-testid="search-form"
  >
    <div
      class="search-form"
    >
      <input
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search Location
      </button>
    </div>
  </div>
  <div
    class="forecast-summaries-app"
    data-testid="forecast-summaries"
  >
    <div
      class="forecast-summaries"
    />
  </div>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  />
</div>
`;

exports[`SearchForm matches snapshot 1`] = `undefined`;
